astroquery/mast/core.py

```diff
@@ -0,0 +1,398 @@
+        
+        return response
+        
+        
+    def _resolve_object(self,objectname,verbose=False):
+        """
+        Resolves an object name to a position on the sky.
+        
+        Parameters
+        ----------
+        objectname : str
+            Name of astronimical object to resolve.
+        verbose : bool (optional)
+            Default False. Setting to True provides more extensive output.    
+        """
+        
+        service = 'Mast.Name.Lookup'
+        params ={'input':objectname,
+                 'format':'json'}
+        
+        response = self.service_request_async(service,params)
+        
+        result = response[0].json() 
+        
+        if len(result['resolvedCoordinate']) == 0:
+            raise ResolverError("Could not resolve %s to a sky position." % objectname)
+            
+        
+        ra = result['resolvedCoordinate'][0]['ra']
+        dec = result['resolvedCoordinate'][0]['decl']
+        coordinates = coord.SkyCoord(ra, dec, unit="deg")
+        
+        return coordinates
+
+
+
+@async_to_sync
+class ObservationsClass(MastClass):
+    """
+    MAST Observations query class.
+
+    Class for querying MAST observational data.
+    """
+    
+        
+    @class_or_instance
+    def query_region_async(self, coordinates, radius="0.2 deg", pagesize=None, page=None, verbose=False):
+        """
+        Given a sky position and radius, returns a list of MAST observations.
+        See column documentation `here <https://masttest.stsci.edu/api/v0/_c_a_o_mfields.html>`_.
+        
+        Parameters
+        ----------
+        coordinates : str or `astropy.coordinates` object
+            The target around which to search. It may be specified as a
+            string or as the appropriate `astropy.coordinates` object. 
+        radius : str or `~astropy.units.Quantity` object (optional)
+            Default 0.2 degrees.
+            The string must be parsable by `astropy.coordinates.Angle`. The
+            appropriate `~astropy.units.Quantity` object from
+            `astropy.units` may also be used. Defaults to 0.2 deg.
+        pagesize : int or None (optional)
+            Default None. 
+            Can be used to override the default pagesize for (set in configs) this query only. 
+            E.g. when using a slow internet connection.
+        page : int or None (optional)
+            Default None.
+            Can be used to override the default behavior of all results being returned to 
+            obtain a sepcific page of results.
+        verbose : bool (optional)
+            Default False. Setting to True provides more extensive output. 
+        
+        
+        Returns
+        -------
+            response: list(`requests.Response`)
+        """       
+        
+        # Put coordinates and radius into consistant format
+        coordinates = commons.parse_coordinates(coordinates)
+        radius = commons.parse_radius(radius.lower())
```

Comment:
``commons.parse_radius`` has been removed, please use ``astropy.coordinates.Angle`` for parsing any angle